mm2src/mm2_bin_lib/build.rs

```diff
@@ -1,114 +1,47 @@
-                if !Regex::new(r"^\w+$").unwrap().is_match(&commit_hash) {
-                    panic!("{}", commit_hash)
-                }
 
-                version = format!("{version}_{commit_hash}");
-            }
-        }
+    let output = Command::new("git")
+        .args(["log", "--pretty=format:%h", "-n1"])
+        .output()
+        .map_err(|e| format!("Failed to run git command: {e}\nSet `KDF_BUILD_TAG` manually instead.",))?;
+
+    let commit_hash = String::from_utf8(output.stdout)
+        .map_err(|e| format!("Invalid UTF-8 sequence: {e}"))?
+        .trim()
+        .to_string();
+
+    if !Regex::new(r"^\w+$")
+        .expect("Failed to compile regex")
+        .is_match(&commit_hash)
+    {
+        return Err(format!("Invalid tag: {commit_hash}"));
     }
 
-    println!("cargo:rustc-env=MM_VERSION={}", version);
-
-    let mut dt_git = None;
-    let mut command = Command::new("git");
-    command.arg("log").arg("--pretty=format:%cI").arg("-n1"); // ISO 8601
-    if let Ok(go) = command.output() {
-        if go.status.success() {
-            let got = from_utf8(&go.stdout).unwrap().trim();
-            let _dt_check = DateTime::parse_from_rfc3339(got).unwrap();
-            dt_git = Some(got.to_string());
-        }
-    }
+    Ok(commit_hash)
+}
 
-    let mm_datetime_p = root().join("../../MM_DATETIME");
-    let dt_file = String::from_utf8(slurp(&mm_datetime_p)).unwrap();
-    let mut dt_file = dt_file.trim().to_string();
-    if let Some(ref dt_git) = dt_git {
-        if dt_git[..] != dt_file[..] {
-            // Create or update the “MM_DATETIME” file in order to appease the Cargo dependency management.
-            let mut mm_datetime_f = fs::File::create(&mm_datetime_p).unwrap();
-            mm_datetime_f.write_all(dt_git.as_bytes()).unwrap();
-            dt_file = dt_git.to_string();
-        }
-    }
+fn version() -> Result<String, String> { version_tag().map(|tag| format!("{}_{}", crate_version(), tag)) }
 
-    println!("cargo:rustc-env=MM_DATETIME={}", dt_file);
+fn build_datetime() -> String { Utc::now().to_rfc3339() }
 
-    version
+fn set_build_variables() -> Result<(), String> {
+    println!("cargo:rustc-env=KDF_VERSION={}", version()?);
+    println!("cargo:rustc-env=KDF_DATETIME={}", build_datetime());
+    Ok(())
 }
 
 fn main() {
-    println!("cargo:rerun-if-env-changed=MANUAL_MM_VERSION");
-    println!("cargo:rerun-if-changed=../../MM_VERSION");
-    println!("cargo:rerun-if-changed=../../MM_DATETIME");
-    if std::env::var("MANUAL_MM_VERSION").is_err() {
-        // This allows build script to run even if no source code files change as rerun-if-changed checks for a file that doesn't exist
-        println!("cargo:rerun-if-changed=NON_EXISTING_FILE");
-    }
-    mm_version();
+    println!("cargo:rerun-if-env-changed=CARGO_PKG_VERSION");
+    println!("cargo:rerun-if-env-changed=KDF_BUILD_TAG");
+
+    set_build_variables().expect("Failed to set build variables");
```

Comment:
Will `set_build_variables` run for local builds after `KDF_BUILD_TAG` is set the first time?

Reply:
It should as we have a check above `cargo:rerun-if-env-changed=KDF_BUILD_TAG`

Reply:
As I understand, If `KDF_BUILD_TAG` is not set manually, the build script will derive it from the latest Git commit hash. After that, if the commit hash changes, the build script will not automatically update `KDF_BUILD_TAG` because of the `cargo:rerun-if-env-changed=KDF_BUILD_TAG` directive, as it only triggers a re-run if the environment variable itself changes, not the underlying Git commit hash.

Reply:
Version build tag worthless if you are actively working on the project. This is only useful to ask people "what version is your application" so we can determine the commit ref in the git history. If they have a local commit that don't exist on our CI, the tag information isn't going to be useful for us.

https://github.com/KomodoPlatform/komodo-defi-framework/pull/2319/commits/9aaa9ae2edee7b88696f2cc785dbb3304edcd6e6 should track the HEAD anyway.

Reply:
> Version build tag worthless if you are actively working on the project. This is only useful to ask people "what version is your application" so we can determine the commit ref in the git history. If they have a local commit that don't exist on our CI, the tag information isn't going to be useful for us.

Agree

> https://github.com/KomodoPlatform/komodo-defi-framework/commit/9aaa9ae2edee7b88696f2cc785dbb3304edcd6e6 should track the HEAD anyway.

It's up to you if you want to keep it or not.